profiler/BaseProfiler.F90

```diff
@@ -100,10 +100,13 @@ subroutine start_self(this, unusable, rc)
       class(KeywordEnforcer), optional, intent(in) :: unusable
       integer, optional, intent(out) :: rc
 
+      !$omp master
       if (this%stack%size()/= 0) this%status = INCORRECTLY_NESTED_METERS
-      _ASSERT_RC(this%stack%size()== 0,"Timer "//this%root_node%get_name()// " is not a fresh self start",INCORRECTLY_NESTED_METERS)
+      !_ASSERT_RC(this%stack%size()== 0,"Timer "//this%root_node%get_name()// " is not a fresh self start",INCORRECTLY_NESTED_METERS)
+      if(this%stack%size() .ne. 0) print*, "Timer "//this%root_node%get_name()// " is not a fresh self start"
```

Comment:
The unit tests rely on an exception being thrown in the bad case, so the PRINT is not an acceptable replacement.    I need to better understand why GFortran rejects the macro inside the OMP region.   Is it the use of any FPP at all?  Or is it the expanded logic and calls?  

If we really can't do the ASSERT then the relevant unit test should be disabled.    I'll have to look up the syntax, but I think  it is something like
`@test`
becomes
`@test(disabled)`

Reply:
Short term fix.  Change the failing `@test` to

`@test(ifndef= __GFORTRAN__)`

(@mathomp4 please check that I've spelled the predefined token correctly.)

Reply:
I agree with @tclune , but in addition  it would be better to uncomment line 176 in profiler/BaseProfile.F90 and comment out line 177 (i.e. we need to throw the correct exception in general)

Reply:
@tclune GFortran error message is:
    _ASSERT_RC(this%stack%size()== 1,"Stack not empty when timer stopped.",INCORRECTLY_NESTED_METERS)
      |                                                                                                                                                                                                                       ^
Error: invalid branch to/from OpenMP structured block